%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
% This is just an example/guide for you to refer to when submitting manuscripts to Frontiers, it is not mandatory to use Frontiers .cls files nor frontiers.tex  %
% This will only generate the Manuscript, the final article will be typeset by Frontiers after acceptance.   
%                                              %
%                                                                                                                                                         %
% When submitting your files, remember to upload this *tex file, the pdf generated with it, the *bib file (if bibliography is not within the *tex) and all the figures.
%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

%%% Version 3.4 Generated 2018/06/15 %%%
%%% You will need to have the following packages installed: datetime, fmtcount, etoolbox, fcprefix, which are normally inlcuded in WinEdt. %%%
%%% In http://www.ctan.org/ you can find the packages and how to install them, if necessary. %%%
%%%  NB logo1.jpg is required in the path in order to correctly compile front page header %%%

\documentclass[utf8]{FrontiersinHarvard} % for articles in journals using the Harvard Referencing Style (Author-Date), for Frontiers Reference Styles by Journal: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal
%\documentclass[utf8]{FrontiersinVancouver} % for articles in journals using the Vancouver Reference Style (Numbered), for Frontiers Reference Styles by Journal: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal
%\documentclass[utf8]{frontiersinFPHY_FAMS} % Vancouver Reference Style (Numbered) for articles in the journals "Frontiers in Physics" and "Frontiers in Applied Mathematics and Statistics" 

%\setcitestyle{square} % for articles in the journals "Frontiers in Physics" and "Frontiers in Applied Mathematics and Statistics" 
\usepackage{url,hyperref,lineno,microtype,subcaption}
\usepackage[onehalfspacing]{setspace}

\linenumbers


% Leave a blank line between paragraphs instead of using \\


\def\keyFont{\fontsize{8}{11}\helveticabold }
\def\firstAuthorLast{Boileau {et~al.}} %use et al only if is more than 1 author
\def\Authors{Etienne Boileau\,$^{1,2,3}$, Xue Li\,$^{2,3}$, Christian Becker$^{4}$, Ramona Casper$^{4}$, Janine Altm{\"u}ller$^{5}$, Florian Leuschner\,$^{2,3}$, Christoph Dieterich\,$^{1,2,3,*}$}
% Affiliations should be keyed to the author's name with superscript numbers and be listed as follows: Laboratory, Institute, Department, Organization, City, State abbreviation (USA, Canada, Australia), and Country (without detailed address information such as city zip codes or street names).
% If one of the authors has a change of address, list the new address below the correspondence details using a superscript symbol and use the same symbol to indicate the author in the author list.
\def\Address{$^{1}$Section of Bioinformatics and Systems Cardiology, Klaus Tschira Institute for Integrative Computational Cardiology, Im Neuenheimer Feld 669, 69120 Heidelberg, Germany
\\
$^{2}$Department of Internal Medicine III (Cardiology, Angiology, and Pneumology), University Hospital Heidelberg, Im Neuenheimer Feld 669, 69120 Heidelberg, Germany
\\
$^{3}$DZHK (German Centre for Cardiovascular Research) Partner Site Heidelberg/Mannheim
\\
$^{4}$Cologne Center for Genomics (CCG), University of Cologne, Weyertal 115b, 50931 Cologne, Germany
\\
$^{5}$Berlin Institute of Health at Charit{\'e}-Universit{\"a}tsmedizin Berlin, Max Delbr{\"u}ck Center for Molecular Medicine 
}
% The Corresponding Author should be marked with an asterisk
% Provide the exact contact address (this time including street name and city zip code) and email of the corresponding author
\def\corrAuthor{Christoph Dieterich}
\def\corrEmail{christoph.dieterich@uni-heidelberg.de}


%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%
\graphicspath{{figures/}}

\usepackage{xspace}

\newcommand{\insi}{\textit{in situ}\xspace}
\newcommand{\exsi}{\textit{ex situ}\xspace}
\newcommand{\apri}{\textit{a priori}\xspace}
\newcommand{\vivo}{\textit{in vivo}\xspace}
\newcommand{\vitro}{\textit{in vitro}\xspace}
\newcommand{\denovo}{\textit{de novo}\xspace}
\newcommand{\ie}{\textit{i.e.}\xspace}
\newcommand{\eg}{\textit{e.g.}\xspace}
\newcommand{\etc}{\textit{etc.}\xspace}
\newcommand{\vs}{\textit{vs.}\xspace}
\newcommand{\etl}{\textit{et al.}\xspace}
\newcommand{\perse}{\textit{per se}\xspace}

\newcommand{\rnaseq}{\textsc{RNA}-seq\xspace}
\newcommand{\scn}{\textsc{scNapBar}\xspace}
\newcommand{\scnast}{\textsc{scNaST}\xspace}

% revision
\usepackage{xcolor}
\newcommand*{\red}{\textcolor{red}}

%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%%

\begin{document}
\onecolumn
\firstpage{1}

\title[Single-cell full-length spatial transcriptomics]{Single-cell full-length spatial transcriptomics reveal the unexplored isoform diversity of the myocardium post-MI} 


\author[\firstAuthorLast ]{\Authors} %This field will be automatically populated
\address{} %This field will be automatically populated
\correspondance{} %This field will be automatically populated

\extraAuth{}% If there are more than 1 corresponding author, comment this line and uncomment the next one.
%\extraAuth{corresponding Author2 \\ Laboratory X2, Institute X2, Department X2, Organization X2, Street X2, City X2 , State XX2 (only USA, Canada and Australia), Zip Code2, X2 Country X2, email2@uni2.edu}


\maketitle


\begin{abstract}
\section{}
We introduce Single-cell Nanopore Spatial Transcriptomics (\scnast), a software suite to facilitate the analysis of spatial gene expression from second- and third-generation sequencing, allowing to generate a full-length single-cell transcriptional landscape of the tissue microenvironment.
Taking advantage of the Visium Spatial platform, we adapted a strategy recently developed to assign barcodes to long-read single-cell sequencing data for spatial capture technology.
Here, we demonstrate our workflow using four short axis sections of the mouse heart following myocardial infarction.
We constructed a \denovo transcriptome using long-read data, and successfully assigned 19,794 transcript isoforms in total, including clinically-relevant, but yet uncharacterized modes of transcription, such as intron retention or antisense overlapping transcription.
We showed a higher transcriptome complexity in the healthy regions, and identified intron retention as a mode of transcription associated with the infarct area.
Our data revealed a clear regional isoform switching among differentially used transcripts for genes involved in cardiac muscle contraction and tissue morphogenesis.
Molecular signatures involved in cardiac remodeling integrated with morphological context may support the development of new therapeutics towards the treatment of heart failure and the reduction of cardiac complications.

\tiny
 \keyFont{ \section{Keywords:} Spatial transcriptomics, Single-cell RNA sequencing, Oxford Nanopore Technology, Myocardial Infarction} 
\end{abstract}

\section*{Introduction}
Cell type heterogeneity has recently emerged as a major aspect in redrawing the cellular picture of the mammalian heart~\citep{Wang2020:human_heart,Tucker2020:human_heart,Litvinukova2020:human_heart}. 
Single-cell RNA-seq (scRNA-seq) technology has enabled to explore crosstalk of different cardiac cell populations to identify response signatures involved in remodeling after myocardial infarction (MI) and ischemic injury~\citep{Cui2020:regenerative_cm,Forte2020:interstitial,Ruiz-Villalba2020:CF,Vafadarnejad2020:cardiac_neutrophil,Molenaar2021:ischemic,Gladka2021:CM_ZEB2,Tombor2021:endothelial_plasticity,Heinrichs2021:B-cell}.
These and other data provide a valuable compendium of information to better understand transcriptional changes occuring in cardiomyocyte and non-cardiomyocyte sub-populations in healthy, injured, and regenerating hearts.
However, in all these studies, the original tissue architecture is destroyed and, in general, the morphological context is lost, including the relationship of cells to infarct, border and remote zones~\citep{Duijvenboden2019:border_zone}.

Recent development in spatial transcriptomics addresses this challenge, but few studies only have provided spatially resolved insights into the cardiac transcriptome.
Techniques such as microdissection~\citep{Wu2016:zebrafish_cm,Burkhard2018:embryonic_heart} or, in particular, spatially barcoded arrays and \insi capturing, have enabled to retain the spatial information while profiling the whole transcriptome at near single-cell resolution, allowing to shed light on localized tissue neighbourhoods~\citep{Asp2017:fetal_markers,Asp2019:spatial_atlas_devel_heart}, but a detailed characterization of cellular zones of injury, repair and/or remodeling is lacking. MI is a complex spatio-temporal heterogeneous disease involving the whole heart, and unbiased spatial transcriptomics holds a promise to add tissue context to molecular profiling in the search of novel therapeutics.

One of the most established spatial transcriptomics methods is now widely available as the Visium platform by 10x Genomics. 
After the tissue section is fixed on the spatial slide, stained, and imaged, it is permeabilized to release RNA to bind to capture probes for on-slide cDNA synthesis.
Library preparation is performed off-slide, and spatial barcodes and tissue image are used to overlay transcriptomics data with tissue context.
Although several new such methods have recently been published~\citep{Rodriques2019:slide-seq,Liu2020:DBiT-seq}, they are for the most part relying on short-read library preparation.
Thus they fail to generate sufficient overlaps to reconstruct transcriptomes \denovo.

Adequate gene and transcript models are instrumental towards relevant proof of concepts and investigational new drug development in translational cardiac research~\citep{Muller2021:pig_transcriptome}. 
%Third-generation or long-read sequencing allow to reconstruct truthful assemblies with fewer gaps and to characterize complete transcript isoforms and chimeric transcripts, at the cost however of lower throughput and poor base-calling accuracy, which may be a problem for cell or spatial barcode assignment.
Third-generation or long-read sequencing allow to reconstruct truthful assemblies with fewer gaps and to characterize complete transcript isoforms and chimeric transcripts.
This has recently been taken to the single-cell level using either Pacific Biosciences (PacBio) or Oxford Nanopore technology~\citep{Gupta2018:ScISOr-Seq,Lebrigand2020:sicelore,Volden2020:R2C2_10x,Wang2021:scnapbar,Joglekar2021:spatial_brain}.

Here, taking advantage of the conceptual similarity between spatial and cell barcodes, we introduce Single-cell Nanopore Spatial Transcriptomics (\scnast), a set of tools to facilitate whole transcriptome spatial profiling of full-length transcripts, based on a previously published Bayesian approach for cell barcode assignment~\citep{Wang2021:scnapbar}. Our method relies on the commercially available Visium platform by 10x Genomics and Nanopore long-read sequencing, although it can in principle be extended to other technologies, as long as a hybrid sequencing approach is used for spatial barcode assignment. 

In this short report, we demonstrate how \scnast can be used to characterize the spatial transcriptional landscape of the heart post-MI.
Using our workflow, we were able to assign a total of 7,616 spatial spots across four short axis sections of the heart, corresponding to distinct 19,794 transcript isoforms in total, encoded by 12,474 genes.
We showed a higher transcriptome complexity in the healthy regions, and identified intron retention as a mode of transcription associated with the injured areas.
Our results showed a clear regional isoform switching among differentially used transcripts for genes involved in cardiac muscle contraction and tissue morphogenesis, many of them clinically relevant, opening new opportunities in translational cardiac research.


\section*{Material and Methods}

\subsection*{Experimental model of myocardial infarction}
A C57BL/6 mouse (female, 8 weeks old, Janvier Labs) was exposed to 5\% isoflurane for anesthesia. An intubation cannula was inserted orally into the trachea. The mouse was fixed on a heating plate at 37$^\circ$C and maintained under anesthesia with 2\% isoflurane. An incision was made from the left sternum to the midclavicular line. Skins and muscle layers were stretched with forceps and sutures. Another incision was made between the third and fourth intercostal space. The heart was exposed and subjected to permanent myocardial infarction with ligation of the left anterior descending (LAD) coronary artery. When the ribs and skins were fully closed, the isoflurane supply was cut off. Oxygen was then supplied until normal breathing was resumed. 

\subsection*{Heart extraction and cryosection}
Three days after permanent LAD ligation, the mouse was sacrificed for organ harvest. After washing with cold PBS several times to remove the blood, the heart was transferred into a bath of isopentane (Millipore Sigma) frozen by liquid nitrogen. The freshly obtained heart was kept fully submerged in isopentane for 5 min until fully frozen. Pre-cooled Cryomold on the dry ice was filled with chilled TissueTek OCT compound without introducing bubbles. The frozen tissue was then transferred into the OCT with pre-cooled forceps and placed on the dry ice until the OCT was completely frozen. OCT-embedded tissue blocks were removed from the Cryomold and mounted on the specimen stage. 10$\mu$m sections were cut in a cryostat at -10$^\circ$C and placed within a Capture Area on the pre-equilibrated Visium Spatial Slides (10x Genomics). The slides were later sealed in individual 50 ml Falcon at -80$^\circ$C ready for further processing. 

\subsection*{10X Genomics Visium experiments}
Four short axis sections of the heart were processed according to the manufacturer's protocol. Libraries were prepared individually, one per heart section. The Visium Spatial Tissue Optimization Slide \& Reagent kit (10x Genomics) was used to optimize permeabilization conditions. Tissue permeabilization was performed for 24 min. Spatially barcoded full-length cDNA was generated using the Visium Spatial Gene Expression Slide \& Reagent kit (10x Genomics). A fraction of each cDNA library was used for Nanopore sequencing. cDNA amplification was then conducted for 20 cycles of PCR (identified by qPCR), and 400$\mu$l were used in the 10xGenomics Visium library preparation (100$\mu$l per section). The libraries were sequenced on a NovaSeq6000 (Illumina), with 29 bases from read 1 and 90 bases from read 2, at a depth of 160M reads per section (640M reads in total). The raw sequencing data was processed with the 10x Genomics Space Ranger v1.1.2 and mapped to the mm10 genome assembly (mm10-2020-A).

\subsection*{Oxford Nanopore sequencing libraries}
Libraries for Nanopore sequencing were prepared according to the manufacturer's protocol for direct cDNA Sequencing (SQK-DCS109 Oxford Nanopore Technologies) with the following minor modifications: The protocol was started with 200ng cDNA with the End-prep step. Final library elution was performed in 15$\mu$l to have material left for TapeStation analysis and Qubit BR measurement. Four GridIon flow cells (FLO-MIN106) were loaded with 12$\mu$L libraries (74-115ng) by using a Flow Cell Priming kit EXP-FLP002. Base calling was done with Guppy v5.0.12. The High accuracy (HAC) model was selected for base calling (Q-Score cut-off $>$ 9).

\subsection*{Spatial barcode assignment}
To account for source-specific quality differences, each heart section (Illumina libraries) was processed separately using Scanpy v1.7.2~\citep{Wolf2018:scanpy}, keeping only spatial barcodes with approximately 150 $<$ counts $<$ 18000, 250 $<$ genes $<$ 5000, detected in at least 2 spots, and with less than 40\% mitochondrial counts. The resulting datasets were concatenated, normalized, and the union of highly variable genes (per section) were kept for final analysis. Batch balanced KNN (BBKNN)~\citep{Polanski2019:bbknn} with ridge regression~\citep{Park2020:thymic_atlas} was used for integration and batch correction, starting from a coarse clustering obtained from a BBKNN-corrected graph. 

For the Nanopore libraries, samples were demultiplexed and processed with ScNapBar v1.1.0 (\url{https://github.com/dieterich-lab/single-cell-nanopore}) using a naive Bayes model to assign spatial barcodes~\citep{Wang2021:scnapbar}.
Briefly, for each heart section, the Space Ranger results (Illumina libraries) were used to parametrize a model of barcode alignment features to discriminate correct versus false barcode assignment in the Nanopore data. 
FASTQ files were mapped using minimap2 v2.21~\citep{Heng2018:minimap}. 
For transcript isoform quantification, a \denovo transcriptome annotation was generated. 
Alignment files were processed by StringTie v2.1.5 in long read mode with the reference annotation to guide the assembly process~\citep{Kovaka2019:stringtie_long}. 
The annotations were merge into a non-redundant set of transcripts and compared to the reference using GffCompare v0.12.2~\citep{Pertea2020:gff}, after removing single-exon transcripts. 
To generate feature-spatial barcode matrices, alignment files were split into multiple files, one per spatial barcode, based on the barcode assignments, converted to FASTQ, and aligned to the \denovo transcriptome with minimap2. 
Abundances were quantified with Salmon v1.5.2 in alignment-based mode using a long read error model~\citep{Patro2017:salmon}. 
Each section was processed separately using Scanpy v1.7.2 and integrated with BBKNN, as described above for the Illumina data.
Spatial barcodes were filtered for counts (approximately 50 $<$ counts $<$ 4000), transcripts (approximately 50 $<$ transcripts $<$ 2000, detected in at least 2 spots), and ribosomal genes ($<$ 40\%).
Neighbors enrichment and cluster co-occurence analyses were performed using Squidpy~\citep{Palla202:squidpy}.

\subsection*{Identification of anatomical regions}
For the Illumina libraries, the neighborhood graph was computed using BBKNN. Spots were clustered with a low resolution (0.3) to identify anatomical regions such as infarct, border and remote zones. Marker genes were identifed using a Wilcoxon rank sum test with Benjamini-Hochberg correction, by comparing the expression of each gene in a given cluster against the rest of the spots. The final clusters were manually annotated.

Labels were then assigned to Nanopore spatial barcodes based on the set of matching Illumina barcodes. 
However, not all assigned barcodes were labeled due to quality control filtering criteria that were different between Illumina and Nanopore datasets. 
To assign labels to the remaining Nanopore barcodes, seed labelling was performed with scANVI using the set of assigned labels as groundtruth~\citep{Xu2021:scANVI}.
The top expressed transcripts were then identified as described above for the Illumina data.

\subsection*{Spatial spot deconvolution}
Spatial spots were deconvoluted using stereoscope (scvi-tools v0.15.0)~\citep{Andersson2020:stereoscope}. Heart data (Smart-Seq2 and 10x Genomics) from the Tabula Muris~\citep{Schaum2018:Tabula_muris} were used as reference dataset and highly variable genes were identified. For the Smart-Seq2 data, gene length normalization was applied. The model was trained on the single cell reference dataset on the intersection of genes found in the spatial (Illumina) data, and proportions were inferred for each Visium spot for each cell type in the reference dataset. Labels were then assigned to Nanopore spots are described above.

\subsection*{Spatial gene expression (Illumina)}
For each heart section, genes with spatial expression patterns were obtained with SPARK (FDR $<$ 0.05)~\citep{Sun2020:spark}. 
Overrepresented biological processes in each region were identified using spatially variable markers that were previously identified (logFC $<$ 0.2, p-value $<$ 0.05), using
Enrichr with the Python package GSEApy.

\subsection*{Differential transcript usage (Nanopore)}
To identify changes in relative usage of transcripts/isoforms within genes, differential transcript usage (DTU) tests were performed using satuRn~\citep{Gilis2021:satuRn}. Only multi-exon transcripts and genes with more than one isoform were kept for the analyses. The transcript count matrix was further filtered to keep transcripts expressed in a worthwhile number of spots, determined by the design (but greater than at least 10\% of the smallest group size), with a CPM count above a threshold of 1(median library size)$^{-1}$. In addition, transcripts were kept only if they had a minimum count of 1 across all spots. A quasi-binomial generalized linear model was fit using a design comparing each anatomical region with another, or each anatomical region with the rest of all regions. A two-stage testing procedure was performed using stageR~\citep{R:stageR}, with an OFDR of 0.05. Results were reported using a student's t-test statistic, computed with estimated log-odds ratios.
To identify isoform-switching genes, significant genes were identifed for each contrast with at least two transcripts showing a switching expression pattern between regions of interest.

\subsection*{Transcript classes}
Transcript classes were assigned in \scnast using Gffcompare~\citep{Pertea2020:gff}.
For the identification of isoforms and transcript classes between regions, markers previously identifed were used to select transcripts in each region with a logFC $>$ 0.1 (p-value $<$ 0.05).
Enrichment of transcripts of a certain class (\eg intron retention) was calculated using a Fisher exact test, using only non-equality classes.
Motif enrichment was performed using Simple Enrichment Analysis (SEA) from the MEME suite, using retained intron sequences \vs annotated (reference transcript) sequences ~\citep{Bailey2015:MEME}.


\section*{Results}

\subsection*{\scnast enables the demarcation of spatially distinct regions of the myocardium post-MI}
Fresh-frozen tissue samples were stained, imaged and fixed on Visium Spatial Gene Expression Slides (10X Genomics) for permeabilization and \insi RNA capture.
Full-length cDNA libraries were split for the preparation of 3' Illumina short-read and direct long-read Nanopore sequencing libraries.
Short-read data were used for the assignment of spatial barcodes to Nanopore reads using the \scn workflow, and subsequently used to define anatomical regions within the tissue organization (Fig.~\ref{fig:1}A).
Long-read data were used for transcriptome assembly and transcript abundance quantification, and layered onto the stained images to reveal the spatial organization of isoform expression.
The Nanopore data comprises of four heart slices (or samples) with a total of 25,5 million reads, reaching a relatively high sequencing saturation (Fig.~\ref{fig:1}B), and providing a significant gain in coverage along full-length transcripts (Fig.~\ref{fig:1}C).
After spatial barcode assignment, libraries had a median of 2.8 million reads per sample, with over 70\% assigned reads (Figs.~\ref{fig:1}D and \nameref{fig:S1}A).
%Nanopore 7616 × 12590, raw = 7616, 19794
%illumina 7708 × 8160, raw = 7708, 14023
Despite variations across samples, we observed a good per spot correlation between Illumina and Nanopore libraries (Fig.~\nameref{fig:S1}B,C).
Per spatial spot (55$\mu$m), after quality filtering, we observed a median read count varying between 593 and 2021 corresponding to a median of 311 (respectively 1000) distincts isoforms (Fig.~\nameref{fig:S2}, see also Figs.~\nameref{fig:S3}~\nameref{fig:S4}).

Clustering based on short-read gene expression defined four broad morphological regions: two remote zones that stemmed from differences in sequencing depth between heart slices, a border zone, and the infarct. 
Remote zones and, to a lesser extent the border zone, were largely associated with cardiomyocyte markers, while the border and infarct zones were characterized by a higher expression of endothelial, myofibroblast, and immune marker genes, as well as with markers of fibrosis and inflammation (Fig.~\ref{fig:2}A).
The region classification was then transfered to the Nanopore data (Fig.~\ref{fig:2}B), and markers of each region were identified (Fig.~\nameref{fig:S5}).
We also investigated, using the Illumina libraries, whether our spatial transcriptomics data reflected known biological processes of myocardial infarction.
We identified spatially variable genes across samples and characterized each region using biological processes (Fig.~\nameref{fig:S6}, and supporting information~\nameref{tab:tabS1}).
Remote zones were generally associated with cardiac muscle contraction linked to an overrepresentation of healthy cardiomyocytes, and included genes such as Strit1 (SERCA regulator DWORF), cardiac troponin I (Tnni3), myosin-binding protein C (Mybpc3), or ventricular myosin light chain-2 (Myl2). 
The border zone was characterized by genes such as Nppb or Ankrd1, both of which were reported to be upregulated in the border zone after MI~\citep{Hama1995:Nppb,Mikhailov2008:Ankrd1}, and other genes associated with the complement system and myogenesis.
Overrepresented processes were however similar to those found in the infarct area.
Top markers of the infarct area included collagens and genes associated with TGF-$\beta$ and p53 signalling, with hypoxia, coagulation, epithelial to mesenchymal transition, and the extracellular matrix.
Overrepresented processes included neutrophil degranulation, and gene expression associated with the innate immune system.
Remote, border, and infarct zones had a clear distinct spatial organization: the two remote zones co-occured at short distances with one another, but did not show any neighborhood enrichment with the border and infarct zones (Fig.~\ref{fig:2}C,D). 
We observed a slight co-enrichment of the infarct and the border zone at medium distances (Fig.~\ref{fig:2}D), revealing how spatial resolution may, in general, affect morphological classification.

Following this strategy, we successfully assigned 7,616 spatial spots, corresponding to distinct 19,794 transcript isoforms in total, encoded by 12,474 genes.
Among all genes, we observed 8,131 (67,3\%) that expressed a single isoform and 3,953 (32.7\%) that expressed 2 or more isoforms (Fig.~\ref{fig:2}E).
Although predicted by our assembly, genes with many isoforms were expressed at a lower threshold and were not included in our final analyses.
We observed variations in the number of isoforms per gene across morphological regions of each heart slice, with significant differences between either of the remote zones and the border and the infarct areas, suggesting the existence of a higher transcriptome diversity in the healthy regions (Fig.~\ref{fig:2}F). 

\subsection*{\scnast reveals the spatial isoform diversity of the myocardium post-MI}
Transcript isoforms were largely associated with exact matches to the reference annotation, multi-exon transcripts with at least one junction match to the reference (\eg exon skipping and exon extension), transcripts longer than the reference (containment of reference), completely novel transcripts (intergenic), transcripts with exonic overlap, intronic transcripts, or retained introns (Fig.~\ref{fig:3}A). 
Among these classes, we were particularly interested in novel transcripts associated with retained introns. Intron retention (IR) occurs when an intron is transcribed into pre-mRNA and remains in the final mRNA. Only recently has IR become of interest due to its associations with complex diseases~\citep{Zhang2020:IR_cancer}.
Interestingly, among differentially expressed transcript markers, we found across the infarct and border zones 65 distinct IR transcripts, compared to 1,612 in the remote zones, corresponding to an odds ratio of 1.73 (p-value 0.008). When considering the infarct area only, the odds were even greater (odds ratio 2, p-value 0.002).
Introns that were retained across the infarct and border zone were enriched in motifs associated with poly(C)-binding protein 1 and 3 (Pcbp1/3), KH RNA-binding domains (Khdrbs1), and Rbm38, a homolog of Rbm24, a pivotal cardiac splice factor~\citep{Weeland2015:AS_heart} (supporting information~\nameref{tab:tabS2}).
The genes harboring these transcripts were enriched in RNA metabolic/catabolic processes, RNA binding, and the unfolded protein response (Fig.~\ref{fig:3}B).
Among these, we found cardiac muscle alpha actin (Actc1), the major protein of the thin filament responsible for cardiac contraction (Fig.~\ref{fig:3}C). 
Our data revealed a relatively high expresion of the IR transcript isoform across all regions, but a comparatively greater contribution to the infarct area (Fig.~\ref{fig:3}D).
At the Actc1 locus, we also identifed two novel antisense overlapping transcripts with a good coverage across all regions (Fig.~\ref{fig:3}C).
There is now growing evidence that antisense-mediated gene regulation is involved in different pathophysiological contexts, including heart disease~\citep{Luther2005:antisense_cardiac,Zinad2017:NATs,Celik2019:Nppa-AS}.
Recently, a long non-coding RNA transcribed antisense near the Actc1 gene was described for its role in cardiomyocyte proliferation and cardiac repair~\citep{Murugavel2019:lncPCR}.
Using the Tabula Muris~\citep{Schaum2018:Tabula_muris} as a reference dataset to perform deconvolution, we identified cardiomyocytes, and to a lesser extent, endothelial and myofibroblasts as the predominant origin of these novel antisense isoforms, with a correlation pattern that matched that of the largest Actc1 annotated isoform (Fig.~\ref{fig:3}E).
%% INTERPRO

\subsection*{\scnast allows to study regional isoform switching with spatial context}
Among the multi-isoform genes, we looked for those showing a splicing pattern variation across regions.
We identified 109 significant regional isoform switching genes across all comparisons between the remote, border, and infarct areas (OFDR 0.05 at both gene and transcript level, supporting information~\nameref{tab:tabS3}).
Our results showed a clear regional isoform switching among differentially used transcripts for genes involved in cardiac muscle contraction and tissue morphogenesis, many of them clinically relevant, such as Ankrd1~\citep{Mikhailov2008:Ankrd1}, Sparc~\citep{McCurdy2011:Sparc}, or Clu~\citep{Turkieh2019:Clu} (Fig.~\ref{fig:4}A).
The largest number of isoform-switching genes were found between the infarct and the remote zones (Fig.~\ref{fig:4}B).
Among them, we found PDZ and LIM domain protein 5 (Pdlim5), a gene encoding for a protein that localizes to the Z-disk by binding to $\alpha$-actinin, and which has been implicated in dilated cardiomyopathy~\citep{Verdonschot2020:Pdlim5} and in heart failure with preserved ejection fraction~\citep{Soetkamp2021:PKC_HFpEF}.
%Titin (Ttn) is an other important protein that binds directly to $\alpha$-actinin in the Z-line.
Pdlim5 undergoes extensive splicing in both human and mouse.
Our data showed a significant change in isoform usage between two annotated protein-coding isoforms between the infarct and the border or the remote zones (Fig.~\ref{fig:4}C,D,E).
The shorter isoform was mostly found in the border and remote zones, while the longer isoform was significantly, albeit at lower levels, expressed in the infarct area (Fig.~\ref{fig:4}C,E).


\section*{Discussion}
\scnast expands the spatial transcriptomics toolbox to third-generation long-read sequencing.
\scnast is based on commercially available platforms, and can be used to investigate the isoform landscape of complex tissue.
We presented here a transcriptome-wide approach to explore annotated and \denovo isoform diversity with morphological context in the mouse heart after myocardial infarction (MI).
Our approach opens-up new opportunities to understand the spatial and molecular organization of the heart following injury.
%The infarcted heart has usually been divided into three zones: the infarct, an ischemic border zone, roughly defined as the area immediately adjacent to the infarct, and a remote zone. 
Although previous studies showed transcriptional differences between proximal and distal areas to the infarct~\citep{Duijvenboden2019:border_zone}, observations were limited to conventional short-read sequencing approaches or were limited in spatial resolution.
Our results not only reflected known biological processes of MI, but also revealed the spatial organization of gene expresion and transcriptome diversity consistent with the underlying biological condition.
We identified novel isoforms in clinically relevant loci and regional isoform switches that could play a role in the myocardium post-MI.
We envision that our work will serve as a reference for future developments and studies integrating long-read sequencing with spatial gene expression data.



\section*{Data Availability}
The datasets generated for this study can be found in the [to be deposited at SRA].
Additional information, supporting data and scripts to generate the figures are available at \url{https://github.com/dieterich-lab/ScNaST}.

\section*{Code Availability}
\scn is publicly available at:\\ \url{https://github.com/dieterich-lab/single-cell-nanopore}.
\scnast is publicly available at:\\ \url{https://github.com/dieterich-lab/ScNaST}.

\section*{Author Contributions}
CD, FL supervised the research and acquired funding.
JA planned and supervised Visium experiments and library preparation.
XL performed MI experiments and tissue extraction.
CB, RC performed Visium experiments and library preparation.
EB analyzed the data and wrote the manuscript.

\section*{Funding}
EB and CD acknowledge support by the Klaus Tschira Stiftung gGmbH [00.219.2013] and by Informatics for Life funded by the Klaus Tschira Foundation. 
CD acknowledges the DZHK (German Centre for Cardiovascular Research) Partner Site Heidelberg/Mannhein.

\section*{Acknowledgments}
EB and CD would like to thank Harald Wilhelmi for maintaining the Dieterich Lab HPC cluster.

\section*{Conflict of Interest}
The authors declare that they have no conflict of interest.

\section*{Supplemental Data}
% \href{http://home.frontiersin.org/about/author-guidelines#SupplementaryMaterial}{Supplementary Material} should be uploaded separately on submission, if there are Supplementary Figures, please include the caption in the same file as the figure. LaTeX Supplementary Material templates can be found in the Frontiers LaTeX folder.

%% Include only the SI item label in the paragraph heading. Use the \nameref{label} command to cite SI items in the text.
\paragraph*{Supplementary Figure S1}
\label{fig:S1}
{\bf \scnast methodology.}

\paragraph*{Supplementary Figure S2}
\label{fig:S2}
{\bf Quality control (Nanopore).}

\paragraph*{Supplementary Figure S3}
\label{fig:S3}
{\bf Quality control (Illumina).}

\paragraph*{Supplementary Figure S4}
\label{fig:S4}
{\bf Spatial distribution of UMIs or counts and genes or transcripts.}

\paragraph*{Supplementary Figure S5}
\label{fig:S5}
{\bf Top markers of each region.}

\paragraph*{Supplementary Figure S6}
\label{fig:S6}
{\bf Enriched processes among spatially variable marker genes.}

%----

\paragraph*{\red{Supplementary Table S1}}
\label{tab:tabS1}
{\bf Spatially variable genes.}

\paragraph*{\red{Supplementary Table S2}}
\label{tab:tabS2}
{\bf Motif enriched in IR transcripts.}

\paragraph*{\red{Supplementary Table S3}}
\label{tab:tabS3}
{\bf Differential transcript usage (isoform-switching genes).}


\bibliographystyle{Frontiers-Harvard} %  Many Frontiers journals use the Harvard referencing system (Author-date), to find the style and resources for the journal you are submitting to: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal. For Humanities and Social Sciences articles please include page numbers in the in-text citations 
%\bibliographystyle{Frontiers-Vancouver} % Many Frontiers journals use the numbered referencing system, to find the style and resources for the journal you are submitting to: https://zendesk.frontiersin.org/hc/en-us/articles/360017860337-Frontiers-Reference-Styles-by-Journal

%\bibliography{main.bib, spatial_trx.bib, ../../../../../../Bartenschlager/analysis/seurat_pipeline/paper/template/single_cell.bib, ../../../../../../Bartenschlager/analysis/seurat_pipeline/paper/template/software.bib}
\bibliography{main.bib}


%%% Make sure to upload the bib file along with the tex file and PDF
%%% Please see the test.bib file for some examples of references

\section*{Figure captions}

%%% Please be aware that for original research articles we only permit a combined number of 15 figures and tables, one figure with multiple subfigures will count as only one figure.
%%% Use this if adding the figures directly in the mansucript, if so, please remember to also upload the files when submitting your article
%%% There is no need for adding the file termination, as long as you indicate where the file is saved. In the examples below the files (logo1.eps and logos.eps) are in the Frontiers LaTeX folder
%%% If using *.tif files convert them to .jpg or .png
%%%  NB logo1.eps is required in the path in order to correctly compile front page header %%%

\begin{figure}[h!]
\begin{center}
\includegraphics[width=\textwidth]{figure1}
\end{center}
\caption{\scnast methodology. A Schematic of the \scnast workflow using a hybrid sequencing approach on Nanopore and Illumina platforms to assign spatial barcodes to long-read sequencing. B Nanopore sequencing saturation showing the number of splice sites detected at various levels of subsampling. A curve that reaches a plateau before getting to 100\% data suggest that all known junctions in the library have been detected. The curve shows the mean $\pm$ SE of four samples. C Normalized transcript coverage for Nanopore and Illumina. The curves show the mean $\pm$ SE of four samples. D Reads assigned by \scn at each step of the workflow. The bars show the mean $\pm$ SE of four samples.}\label{fig:1}
\end{figure}


\begin{figure}[h!]
\begin{center}
\includegraphics[width=\textwidth]{figure2}
\end{center}
\caption{Defining morphological regions after MI. A Dot plot showing the expresson of selected markers associated with the expresson of CM=cardiomyocytes, EC=endothelial cells, MFB=myofibroblasts, IM=immune cells, or with fibrosis and inflammation, based on the short-read Illumina data. B Annotation of mouse heart regions after MI via short-read clustering, transfered to the Nanopore data. Scatter plot in spatial coordinates of the anatomical regions (left) and UMAP representation of the Nanopore data using the region annotation from short-read clustering (right). Colors in the spatial scatter plot are matching those of the UMAP. C Neighbors enrichment analysis in one heart axis section. The heatmap shows the enrichment score on spatial proximity between the different anatomical regions. Spots belonging to two different regions that are close together will have a high score, and vice-versa. D Cluster co-occurrence in spatial dimensions in one heart axis section. Line plot showing the conditional probability of observing a given region conditioned on the presence of the infarct region, computed across increasing radii size around each spots. Distance units are given in pixels of the Visium source image. E Barplot showing the frequency distribution of the number of isoforms per gene, either stemming from the assembly, or found in the final data after quality control filtering. The median length of transcripts is indicated in each bar for each category. F Average number of isoform per gene detected among markers of each morphological region. Significance was measured using a Mann-Whitney U-test (*** = $<$0.001, **** = $<$0.0001)}\label{fig:2}
\end{figure}


\begin{figure}[h!]
\begin{center}
\includegraphics[width=\textwidth]{figure3}
\end{center}
\caption{Characterizing the isoform diversity after MI. A Barplot showing how full-length transcripts obtained with \scnast compare to the existing mouse annotation. Labels Complete (=), Multi-exon (j), Containment (k), Unknown (u), Exonic (x) , Intron(i), Overlap (o), Retained (m, n), and Contains (y) are explained in \url{https://ccb.jhu.edu/software/stringtie/gffcompare.shtml}. B Over-representation analysis of genes harboring novel transcripts with intron retention (IR). From top to bottom, biological processes, molecular function, and hallmark gene sets from the Molecular Signatures Database (MSigDB). C Exonic structure of the different Actc1 isoforms, including novel isoforms identified by \scnast with IR and exonic antisense overlap. Coverage (log scale) is shown for each region with a different scale. D Actc1 isoform contributions to total Actc1 expression in the different heart regions. E Per spot correlation observed between spatial deconvolution of cell types and Actc1 isoforms. }\label{fig:3}
\end{figure}


\begin{figure}[h!]
\begin{center}
\includegraphics[width=\textwidth]{figure4}
\end{center}
\caption{Regional isoform switching after MI. A Dot plot showing the top 5 most significant isoform-switching genes per contrast across all comparisons between the remote, border, and infarct areas, using a stage-wise testing procedure at an overall false discovery rate (OFDR) of 0.05. The top genes were restricted to the spatially variable genes. B Heatmap representing the number of isoform-switching genes identified between any two regions. I=infarct, BZ=border zone, RZ=remote zone. C Spatial scatter plot showing Pdlim5 isoform expression in one heart axis section. D Pdlim5 isoforms track. E Zoom of D to show coverage of both isoforms across regions.}\label{fig:4}
\end{figure}



\end{document}
